src/catalog/src/durable/impls/migrate.rs

```diff
@@ -211,6 +229,9 @@ impl CatalogMigrator {
                 Self::rollback_from_persist_to_stash(&mut stash, persist).await?;
                 Ok(stash)
             }
+            Direction::EmergencyStash => {
+                unreachable!("handled earlier without opening persist")
+            }
```

Comment:
It's up to you, but I think it would be better to set up the API for this function in such a way that it's impossible to get here. For example, by taking `is_to_persist: bool` instead of `direction: Direction`, or by changing the structure of the `Direction` type, e.g.

```
enum RealDirection {
    MigrateToPersist,
    RollbackToStash,
}
enum Direction {
    EmergencyStash,
    RealDirection(RealDirection),
}
```

(those names I made up are terrible but you get the idea)

Reply:
If you don't want to do that, can you at least add a comment to this function explaining that it can't be called with `Direction::EmergencyStash` ?

Reply:
I think you're approach is much better, I was just being lazy. I've updated it to the two enum approach.